### ISSUE TITLE
Does anyone know why dependency track is unable to identify CVEs in hardware components despite entering the correct CPE? CPE for hardware components does not include the version number. Can someone help?

### ISSUE THREAD

**Issue by chinmaybhave05** [DependencyTrack/dependency-track#1320]
All questions should be directed to the Dependency-Track Slack channel or to GitHub discussions
Questions submitted through GitHub issues will be closed.

Slack channel
  - https://dependencytrack.org/slack

Slack invite
  - https://dependencytrack.org/slack/invite
  
GitHub discussions
  - https://github.com/DependencyTrack/dependency-track/discussions

Groups.io (mailing list)
  - https://dependencytrack.org/discussion

Providing as much detail as possible, what is your question?

**Comment by chinmaybhave05** [comment]
The tool works fine with identifying CVEs for software components but does not work for hardware components. I think the logic to parse CPE is not consistent with the data published by NVD which is causing this issue. Is anyone else facing the same?

**Comment by stevespringett** [comment]
I regularly test with hardware CPEs as well. I quickly found the first hardware CPE I could find and added a component to it. DT was able to properly identify vulnerabilities for the component.

<img width="1722" alt="Screen Shot 2022-01-06 at 11 03 00 AM" src="https://user-images.githubusercontent.com/3878933/148421441-6fb111ab-0486-4ad8-8439-fe64ba2d5e7e.png">
.

**Comment by chinmaybhave05** [comment]
Yes, it works well with CPEs which have vendor name, component name and version number. However, it fails where a CPE does not have one of the three items. For e.g. it does not work well with CPE - cpe:2.3:h:intel:2000e:-:*:*:*:*:*:*:* 

Below NVD search shows CVEs while DT is unable to identify CVEs.

https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&isCpeNameSearch=true&seach_type=all&query=cpe:2.3:h:intel:2000e:-:*:*:*:*:*:*:*

**Comment by chinmaybhave05** [comment]
![DT](https://user-images.githubusercontent.com/71174866/148730240-a35c1f04-e884-443c-b75d-5509d673cbdf.PNG)

**Comment by stevespringett** [comment]
Thanks for the clarification.

I found the issue and have committed a fix. This will be available in DT 4.4.

One note however, is that `cpe:2.3:h:intel:2000e:-:*:*:*:*:*:*:*` itself isn't vulnerable according to the data in CVE-2019-0174. However, `cpe:2.3:o:intel:2000e_firmware:-:*:*:*:*:*:*:*` is vulnerable and with this fix, DT now finds finds the vulnerability.

```json
{
  "operator" : "OR",
  "children" : [ ],
  "cpe_match" : [ 
    {
      "vulnerable" : false,
      "cpe23Uri" : "cpe:2.3:h:intel:2000e:-:*:*:*:*:*:*:*",
      "cpe_name" : [ ]
    }
  ]
}
```

**Comment by chinmaybhave05** [comment]
Thanks for acknowledging it as an issue. I hope this helps make the tool better.

Another point here is, mapping of CVEs to a component is based on the CPE entered. In case of hardware components, version numbers are not always added to a CPE whereas version number is a mandatory field in DT. I think it is a good idea to add a checker to segregate between a hardware or software component or making version number field optional should resolve this issue.

Also, I am trying to setup this tool with the hardware vulnerability tracking capability now. Releasing v4.4 with the fix might take time. Is it possible to share details of what needs to be changed in the current version to resolve this issue or release a patch or something? I believe you have already identified and fixed the issue.
